model/domain/signedpiece.go

```diff
@@ -6,25 +6,38 @@ import (
 )
 
 type SignedPiece struct {
-	Piece     *Piece
-	Signature *Signature
+	pieceSerial []byte
```

Comment:
move pieceSerial to Piece model, rename to bytes and add method AsBytes in Piece? What do you think? Also Piece can have constructor method FromBytes

Reply:
Yes that is the other option. I thought this feature is more about signed pieces because that’s the structure that is actually transmitted or stored or verified.